test/test_pynisher.py

```diff
@@ -479,6 +479,33 @@ def test_raise(self):
             self.assertEqual(wrapped_function.os_errno, 2)
         self.assertEqual(wrapped_function.exitcode, 0)
 
+    def test_capture_output_error(self):
+        grace_period = 1
+
+        # We want to mimic an scenario where the context.Pipe
+        # fails early, so that a stdout file was not created.
+        context = unittest.mock.Mock()
+        logger_mock = unittest.mock.Mock()
+        context.Pipe.return_value = (unittest.mock.Mock(), unittest.mock.Mock())
+        context.Pipe.return_value[0]._side_effect = ValueError()
+
+        wrapped_function = pynisher.enforce_limits(
+            wall_time_in_s=1,
+            mem_in_mb=None,
+            context=context,
```

Comment:
So the context is a mock, which means no subprocess is started, right? Would it be possible to use a real context and mock out only the Pipe?

Reply:
This, I do not know how to do. Because part of the trick is to crash the function quickly.

I tried only mocking the method via a `mock.patch.object` and by overwriting the method. This triggered the subprocess to run, but in turn, allowed the function to execute with enough time to create the file. Hence the test fails.

So in other words, to reproduce, we must also fail in the subprocess function, get to the try block and fail there without writing to the stdout.

Reply:
I see. I also wouldn't know how to do this, so let's merge this one.